src/main/java/com/microsoft/spring/data/gremlin/query/GremlinTemplate.java

```diff
@@ -63,4 +67,25 @@ public void deleteAll() {
 
         return object;
     }
+
+    @Override
+    public <T> T findVertexById(@NonNull Object id, @NonNull Class<T> domainClass) {
+        final Client client = this.gremlinFactory.getGremlinClient();
+        final String idValue = id.toString();
+        final GremlinSource source = new GremlinSourceVertex(idValue, new GremlinResultVertexReader());
+        final GremlinScript<String> script = new GremlinScriptVertexFindByIdLiteral();
+        final List<Result> results;
+
+        try {
+            results = client.submit(script.generateScript(source)).all().join();
+        } catch (RuntimeException e) {
+            return null; // If cannot find the vertex by Id, ignore exception and return null
```

Comment:
Does that mean if there is exception thrown, it can only be caused by there is no vertex for given `id`? No other cases will throw RuntimeException?

Reply:
good question, I may need to make some mistakes to figure out that. I am a little concern about that it hidden all exception and just throw runtimeException outside.

Reply:
Updated, and handle the CompletionException. My mistake that if no `id` found, it only return null instead of exception. Double checked.

Reply:
Thanks and sorry for not mentioning this earlier.

Wonder whether or not we should wrap our self defined exceptions to align with Spring data exceptions, i.e., extend from Spring Data exceptions to allow users to switch between different Spring Data modules.

https://docs.spring.io/spring/docs/current/spring-framework-reference/data-access.html#dao-exceptions

`"....This allows one to switch between the aforementioned persistence technologies fairly easily and it also allows one to code without worrying about catching exceptions that are specific to each technology."`

Reply:
Updated and extends from spring data.